src/workspace/views/Bookmarks.tsx

```diff
@@ -1,13 +1,270 @@
+import { LoadingInfo } from '../../shared/components/LoadingErrorLoadedComponent/LoadingErrorLoadedComponent';
+import { buildLink, CustomError, formatDate, formatTimestamp, fromNow } from '../../shared/helpers';
+import { BookmarksViewsPlaysService } from '../../shared/services/bookmarks-views-plays-service';
+import { BookmarkInfo } from '../../shared/services/bookmarks-views-plays-service.const';
+import toastService from '../../shared/services/toast-service';
+
+const ITEMS_PER_PAGE = 5;
+
+interface BookmarksOverviewProps extends DefaultSecureRouteProps {
+	numberOfItems: number;
+}
+
+const BookmarksOverview: FunctionComponent<BookmarksOverviewProps> = ({
+	numberOfItems,
+	history,
+	user,
+}) => {
+	const [t] = useTranslation();
+
+	// State
+	const [bookmarks, setBookmarks] = useState<BookmarkInfo[]>([]);
+	const [bookmarkToDelete, setBookmarkToDelete] = useState<BookmarkInfo | null>(null);
+	const [isDeleteModalOpen, setIsDeleteModalOpen] = useState<boolean>(false);
+	const [sortColumn, setSortColumn] = useState<keyof BookmarkInfo>('createdAt');
+	const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('desc');
+	const [page, setPage] = useState<number>(0);
+	const [loadingInfo, setLoadingInfo] = useState<LoadingInfo>({ state: 'loading' });
+	const [paginatedBookmarks, setPaginatedBookmarks] = useState<BookmarkInfo[]>([]);
+
+	const fetchBookmarks = useCallback(async () => {
+		try {
+			const bookmarkInfos = await BookmarksViewsPlaysService.getAllBookmarksForUser(user);
+			setBookmarks(bookmarkInfos);
+			setLoadingInfo({ state: 'loaded' });
+		} catch (err) {
+			console.error(new CustomError('Failed to get all bookmarks for user', err, { user }));
+			setLoadingInfo({ state: 'error', message: t('Het ophalen van je bladwijzers is mislukt') });
+		}
+	}, [user, setBookmarks, setLoadingInfo, t]);
+
+	useEffect(() => {
+		fetchBookmarks();
+	}, [fetchBookmarks]);
+
+	const updatePaginatedBookmarks = useCallback(() => {
+		setPaginatedBookmarks(
+			orderBy(bookmarks, [sortColumn], [sortOrder]).slice(
+				ITEMS_PER_PAGE * page,
+				ITEMS_PER_PAGE * (page + 1)
+			)
+		);
+	}, [setPaginatedBookmarks, sortColumn, sortOrder, page, bookmarks]);
+
+	useEffect(updatePaginatedBookmarks, [updatePaginatedBookmarks]);
+
+	const onDeleteBookmark = async () => {
+		try {
+			if (!bookmarkToDelete) {
+				toastService.danger(t('Het verwijderen van de bladwijzer is mislukt'));
+				return;
+			}
+			await BookmarksViewsPlaysService.toggleBookmark(
+				bookmarkToDelete.contentId,
+				user,
+				bookmarkToDelete.contentType,
+				true
+			);
+
+			await fetchBookmarks();
+			toastService.success(t('de bladwijzer is verwijderd'));
+		} catch (err) {
+			console.error(new CustomError('Failed t delete bookmark', err, { bookmarkToDelete, user }));
+			toastService.danger(t('Het verwijderen van de bladwijzer is mislukt'));
+		}
+
+		setBookmarkToDelete(null);
+	};
+
+	const onClickColumn = (columnId: keyof BookmarkInfo) => {
```

Comment:
Not necessarily dringend, maar misschien een TODO toevoegen dat we eens kijken of we hiervan een helper-functie kunnen of een sort hook of iets dergelijks want we gebruiken dit in een hoop verschillende pagina's.